client/packages/programs/src/RnRForms/DetailView/Footer.tsx

```diff
@@ -6,17 +6,38 @@ import {
   useBreadcrumbs,
   useTranslation,
   RnRFormNodeStatus,
+  useNotification,
 } from '@openmsupply-client/common';
 import { useRnRForm } from '../../api';
 
-export const Footer = ({ rnrFormId }: { rnrFormId: string }) => {
-  const t = useTranslation();
+export const Footer = ({
+  rnrFormId,
+  linesUnconfirmed,
+}: {
+  rnrFormId: string;
+  linesUnconfirmed: boolean;
+}) => {
+  const t = useTranslation('programs');
   const { navigateUpOne } = useBreadcrumbs();
+  const { error, info, success } = useNotification();
   const {
     query: { data },
     finalise: { finalise, isFinalising },
   } = useRnRForm({ rnrFormId });
 
+  const onFinalise = async () => {
+    if (linesUnconfirmed) {
+      info(t('messages.all-lines-must-be-confirmed'))();
```

Comment:
This isn't quite ideal, requiring them to click through and confirm each line...

Probably should have a dirty state instead, if any lines are dirty, prevent finalising (and warn on refresh/navigate away) - this felt a little more complex given the edit state is managed on a per line basis, so was hoping to do this in another PR...